test/e2e/apigatewayv2/tests/test_api.py

```diff
@@ -0,0 +1,539 @@
+
+    # add permissions for apigateway to invoke authorizer lambda
+    authorizer_arn = "arn:aws:execute-api:{region}:{account}:{api_id}/authorizers/{authorizer_id}".format(
+        region=get_aws_region(),
+        account=get_aws_account_id(),
+        api_id=test_resource_values['API_ID'],
+        authorizer_id=authorizer_id
+    )
+    lambda_client = boto3.client("lambda")
+    lambda_client.add_permission(FunctionName=get_bootstrap_resources().AuthorizerFunctionName,
+                                 StatementId='apigatewayv2-authorizer-invoke-permissions',
+                                 Action='lambda:InvokeFunction',
+                                 Principal='apigateway.amazonaws.com',
+                                 SourceArn=authorizer_arn)
+
+    yield authorizer_ref, cr
+
+    k8s.delete_custom_resource(authorizer_ref)
+    time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+
+@pytest.fixture(scope="module")
+def route_resource(integration_resource, authorizer_resource):
+    route_resource_name = test_resource_values['ROUTE_NAME']
+    route_ref, route_data = helper.route_ref_and_data(route_resource_name=route_resource_name,
+                                                      replacement_values=test_resource_values)
+    if k8s.get_resource_exists(route_ref):
+        pytest.fail(f"expected {route_resource_name} to not exist. Did previous test cleanup?", False)
+    logging.debug(f"apigatewayv2 route resource. name: {route_resource_name}, data: {route_data}")
+
+    k8s.create_custom_resource(route_ref, route_data)
+    cr = k8s.wait_resource_consumed_by_controller(route_ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(route_ref)
+
+    route_id = cr['status']['routeID']
+    test_resource_values['ROUTE_ID'] = route_id
+
+    yield route_ref, cr
+
+    k8s.delete_custom_resource(route_ref)
+    time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+
+@pytest.fixture(scope="module")
+def stage_resource(route_resource):
+    stage_resource_name = test_resource_values['STAGE_NAME']
+    stage_ref, stage_data = helper.stage_ref_and_data(stage_resource_name=stage_resource_name,
+                                                      replacement_values=test_resource_values)
+    if k8s.get_resource_exists(stage_ref):
+        pytest.fail(f"expected {stage_resource_name} to not exist. Did previous test cleanup?", False)
+    logging.debug(f"apigatewayv2 stage resource. name: {stage_resource_name}, data: {stage_data}")
+
+    k8s.create_custom_resource(stage_ref, stage_data)
+    cr = k8s.wait_resource_consumed_by_controller(stage_ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(stage_ref)
+
+    yield stage_ref, cr
+
+    k8s.delete_custom_resource(stage_ref)
+    time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+
+@service_marker
+@pytest.mark.canary
+class TestApiGatewayV2:
+
+    def test_perform_invocation(self, api_resource, stage_resource):
+        time.sleep(DEPLOYMENT_WAIT_AFTER_SECONDS)
```

Comment:
What's the purpose of this sleep? We have already asserted, in the fixture ,that `k8s.wait_resource_consumed_by_controller`?

Reply:
> What's the purpose of this sleep? We have already asserted, in the fixture ,that `k8s.wait_resource_consumed_by_controller`?

This is APIGateway specific wait. Where we wait for changes to get propagated to apigateway dataplane before performing invocation.